src/panels/config/helpers/forms/ha-schedule-form.ts

```diff
@@ -208,16 +208,22 @@ class HaScheduleForm extends LitElement {
   private get _events() {
     const events: any[] = [];
     const currentDay = new Date().getDay();
-    const baseDay = currentDay === 0 ? 7 : currentDay;
+    const baseDay =
+      currentDay === 0 && firstWeekdayIndex(this.hass.locale) === 1
+        ? 7
```

Comment:
We should fix this better, as firstWeekDay could also be on saturday

Reply:
Yes this is not correct.

Reply:
> We should fix this better, as firstWeekDay could also be on saturday

Yes, Saturday was not considered. I am trying that right now, but haven't gotten a solution yet.

Reply:
Will be covered by https://github.com/home-assistant/frontend/pull/14566